cytosnake/utils/cytosnake_setup.py

```diff
@@ -0,0 +1,120 @@
+"""
+Module: cytosnake_setup.py
+
+This modules sets up the current directory as a project directory. Project
+directories are dictated by the presence of `.cytosnake`, which contains 
+meta data for cytosnake to use. 
+
+"""
+import json
+import shutil
+
+from pathlib import Path
+from cytosnake.utils.cyto_paths import (
+    get_cytosnake_package_path,
+    get_config_fpaths,
+    get_workflow_fpaths,
+    get_project_dirpaths,
+)
+
+from cytosnake.common.errors import display_error
+
+
+def create_cytosnake_dir() -> None:
+    """Sets up a `.cytosnake` folder that contains meta information that helps
+    cytosnake's cli functionality. This will include configurations that
+    contains states, paths and settings.
+
+
+    Returns
+    -------
+    None
+        Generates a `.cytosnake` file in current directory
+
+    Raises:
+    FileExistsError
+        Raised by pathlib.Path if `.cytosnake' exists in cur
+    ."""
+
+    # create the `.cytosnake` folder in where `cytosnake init` mode was executed
+    # - if the current director has a `.cytosnake` FileExistError will be raised
+    cwd = Path().absolute()
+    project_folder_path = cwd / ".cytosnake"
+    try:
+        project_folder_path.mkdir(exist_ok=False)
+    except FileExistsError as e:
+        msg = "'.cytosnake' directory exists in current directory"
+        display_error(e, msg=msg)
```

Comment:
This pattern seems to be repeated a bit - I wonder if you can make a generic function to perform. (Discussion point!)

Reply:
That's currently on the works. Probably might see this change in the `helper_functions` PR

Reply:
This sounds like a good option to create an issue?

Reply:
Created issue #29. Will add this to my checklist when developing `helper_fuction` PR